pkg/build/oci/index.go

```diff
@@ -54,8 +54,27 @@ func GenerateDockerIndex(ctx context.Context, ic types.ImageConfiguration, imgs
 
 // generateIndexWithMediaType generates an index or docker manifest list from the given imgs. The index type
 // is provided by the `mediaType` parameter.
-func generateIndexWithMediaType(mediaType ggcrtypes.MediaType, _ types.ImageConfiguration, imgs map[types.Architecture]oci.SignedImage) (name.Digest, oci.SignedImageIndex, error) {
-	idx := signed.ImageIndex(mutate.IndexMediaType(empty.Index, mediaType))
+func generateIndexWithMediaType(mediaType ggcrtypes.MediaType, ic types.ImageConfiguration, imgs map[types.Architecture]oci.SignedImage) (name.Digest, oci.SignedImageIndex, error) {
+	// If annotations are set and we're using the OCI mediaType, set annotations on the index.
+	annotations := map[string]string{}
+	if mediaType == ggcrtypes.OCIImageIndex {
+		annotations = ic.Annotations
+		if annotations == nil {
+			annotations = map[string]string{}
+		}
+		if ic.VCSUrl != "" {
+			if url, hash, ok := strings.Cut(ic.VCSUrl, "@"); ok {
+				annotations["org.opencontainers.image.source"] = url
+				annotations["org.opencontainers.image.revision"] = hash
```

Comment:
Resurfacing Jon's question about how this makes the image churn.

Reply:
I don't follow this - if the hash changes hasn't the image changed?

Reply:
If you change the README in the repo you'll get a new source commit, without changing anything about the image. If you have a repo with dozens of apko YAMLs you don't want to have to rebuild all of them when you change any of them, etc.

I think we're fine with this as-is because `VCSUrl` can be disabled, and is disabled in chainguard-images/images, but I think it's worth considering what we want the default to be for regular apko users.

I'm also fine punting on the question to land this PR.

Reply:
> I think we're fine with this as-is because VCSUrl can be disabled, and is disabled in chainguard-images/images

This addresses my concern.